x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/ReplaceDataStreamBackingIndexStepTests.java

```diff
@@ -61,12 +61,13 @@ public void testPerformActionThrowsExceptionIfIndexIsNotPartOfDataStream() {
             IndexMetadata.builder(indexName).settings(settings(Version.CURRENT).put(LifecycleSettings.LIFECYCLE_NAME, policyName))
                 .numberOfShards(randomIntBetween(1, 5)).numberOfReplicas(randomIntBetween(0, 5));
 
+        final IndexMetadata sourceIndexMetadata = sourceIndexMetadataBuilder.build();
         ClusterState clusterState = ClusterState.builder(emptyClusterState()).metadata(
-            Metadata.builder().put(sourceIndexMetadataBuilder).build()
+            Metadata.builder().put(sourceIndexMetadata, false).build()
```

Comment:
A neat side effect of this cleanup is that it made it easier to catch this bug. We were calling the builder twice, but the second invocation would actually produce a broken result because it created an immutable open map with a `null` map inside that would NPE on every operation if used.